commons/src/test/java/org/infinispan/commons/marshall/JavaSerializationMarshallerTest.java

```diff
@@ -0,0 +1,84 @@
+package org.infinispan.commons.marshall;
+
+import static junit.framework.TestCase.assertTrue;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+
+import java.io.Serializable;
+
+import org.infinispan.commons.CacheException;
+import org.junit.Assert;
+import org.junit.Test;
+
+public class JavaSerializationMarshallerTest {
+   @Test
+   public void testPrimitivesAndArrays() throws Exception {
+      JavaSerializationMarshaller marshaller = new JavaSerializationMarshaller();
+      isMarshallable(marshaller, byte.class);
+      isMarshallable(marshaller, byte[].class);
+      isMarshallable(marshaller, Byte.class);
+      isMarshallable(marshaller, Byte[].class);
+
+      isMarshallable(marshaller, short.class);
+      isMarshallable(marshaller, short[].class);
+      isMarshallable(marshaller, Short.class);
+      isMarshallable(marshaller, Short[].class);
+
+      isMarshallable(marshaller, int.class);
+      isMarshallable(marshaller, int[].class);
+      isMarshallable(marshaller, Integer.class);
+      isMarshallable(marshaller, Integer[].class);
+
+      isMarshallable(marshaller, long.class);
+      isMarshallable(marshaller, long[].class);
+      isMarshallable(marshaller, Long.class);
+      isMarshallable(marshaller, Long[].class);
+
+      isMarshallable(marshaller, float.class);
+      isMarshallable(marshaller, float[].class);
+      isMarshallable(marshaller, Float.class);
+      isMarshallable(marshaller, Float[].class);
+
+      isMarshallable(marshaller, double.class);
+      isMarshallable(marshaller, double[].class);
+      isMarshallable(marshaller, Double.class);
+      isMarshallable(marshaller, Double[].class);
+
+      isMarshallable(marshaller, char.class);
+      isMarshallable(marshaller, char[].class);
+      isMarshallable(marshaller, Character.class);
+      isMarshallable(marshaller, Character[].class);
+
+      isMarshallable(marshaller, boolean.class);
+      isMarshallable(marshaller, boolean[].class);
+      isMarshallable(marshaller, Boolean.class);
+      isMarshallable(marshaller, Boolean[].class);
```

Comment:
Wow the GH highlighting of Boxed primitive arrays is super confusing :upside_down_face: